src/connectors/autocomplete/connectAutocomplete.js

```diff
@@ -0,0 +1,153 @@
+ * @property {string} currentRefinement The actual value of the query.
+ * @property {Object} widgetParams All original widget options forwarded to the `renderFn`.
+ */
+
+/**
+ * @typedef {Object} CustomAutocompleteWidgetOptions
+ * @property {string[]} [indices = []] Name of the others indices to search into.
+ * @property {boolean} [escapeHits = false] If true, escape HTML tags from `hits[i]._highlightResult`.
+ */
+
+/**
+ * **Autocomplete** connector provides the logic to build a widget that will give the user the ability to search into multiple indices.
+ *
+ * This connector provides a `refine()` function to search for a query and a `currentRefinement` as the current query used to search.
+ *
+ * THere's a complete example available on how to write a custom **Autocomplete** widget:
+ * [autocomplete.js](https://github.com/algolia/instantsearch.js/blob/develop/dev/app/custom-widgets/jquery/autocomplete.js)
+ * @type {Connector}
+ * @param {function(AutocompleteRenderingOptions, boolean)} renderFn Rendering function for the custom **Autocomplete** widget.
+ * @param {function} unmountFn Unmount function called when the widget is disposed.
+ * @return {function(CustomAutocompleteWidgetOptions)} Re-usable widget factory for a custom **Autocomplete** widget.
+ */
+export default function connectAutocomplete(renderFn, unmountFn) {
+  checkRendering(renderFn, usage);
+
+  return (widgetParams = {}) => {
+    const { indices } = widgetParams;
+
+    if (indices && !Array.isArray(indices)) {
+      throw new Error(usage);
+    }
+
+    return {
+      getConfiguration() {
+        return widgetParams.escapeHits ? tagConfig : undefined;
+      },
+
+      init({ instantSearchInstance, helper }) {
+        this._refine = this.refine(helper);
+
+        this.indices = [
+          {
+            helper,
+            label: 'primary',
+            index: helper.getIndex(),
+            results: undefined,
+          },
+        ];
+
+        // add additionnal indices into `this.indices`
+        if (indices && Array.isArray(indices)) {
+          indices.forEach(({ label, value }) => {
+            const derivedHelper = helper.derive(searchParameters =>
+              searchParameters.setIndex(value)
+            );
+
+            this.indices.push({
+              label,
+              index: value,
+              helper: derivedHelper,
+              results: undefined,
+            });
+
+            // update results then trigger render after a search from any helper
+            derivedHelper.on('result', results =>
+              this.saveResults({ results, index: value })
+            );
+          });
+        }
+
+        this.instantSearchInstance = instantSearchInstance;
+        this.renderWithAllIndices({ isFirstRendering: true });
+      },
+
+      saveResults({ results, index }) {
+        const derivedIndex = this.indices.find(i => i.index === index);
```

Comment:
By using the name of the index as key we can't have an autocomplete with the same index multiple times. Right now it's useless because we can't provide additional search parameters. But we can image to implement this functionality in the future. We have the issue with React InstantSearch, the idea was to require a `key` to avoid relying on the index name.

Reply:
We have a key called `label` as well, I will switch the find using this key it will solve the issue 👍

Reply:
We can add a message in the doc who specifies that this value must be unique, since we are using it for retrieve the correct index.